cmd/libsnap-confine-private/feature.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SNAP_CONFINE_FEATURE_H
+#define SNAP_CONFINE_FEATURE_H
+
+#include <stdbool.h>
+
+typedef enum sc_feature_flag {
+	SC_PER_USER_MOUNT_NAMESPACE,
+} sc_feature_flag;
+
+/**
+ * sc_feature_enabled returns true if a given feature flag has been activated
+ * by the user via "snap set core experimental.xxx=true". This is determined by
+ * testing the presence of a file in /var/lib/snapd/features/ that is named
+ * after the flag name.
+**/
+bool sc_feature_enabled(sc_feature_flag flag);
```

Comment:
I thought about this but I decided to use an enum instead. There are only so many feature flags and it is easier to cause a typo this way (because you can always pass a string manually). This has the advantage that the code knows the set of flags that exist so it cannot be fooled.

Reply:
+1 for typos